addons/mail/static/src/core/common/mention_model.js

```diff
@@ -0,0 +1,71 @@
+import { OR, Record } from "@mail/core/common/record";
+import { _t } from "@web/core/l10n/translation";
+
+export class Mention extends Record {
+    static id = OR("channel", "partner", "special");
+    /** @type {Object.<number, import("models").Mention>} */
+    static records = {};
+    /** @returns {import("models").Mention} */
+    static get(data) {
+        return super.get(data);
+    }
+    /** @returns {import("models").Mention|import("models").Mention[]} */
+    static insert(data) {
+        return super.insert(...arguments);
+    }
+    static new() {
+        return super.new(...arguments);
+    }
+
+    static specialMentions = {
+        everyone: {
+            label: "everyone",
+            channel_types: ["channel", "group"],
+            displayName: "Everyone",
+            description: _t("Notify everyone"),
+        },
+    };
+
+    channel = Record.one("Thread");
+    partner = Record.one("Persona");
+    /** @type { 'everyone' | 'here' | 'admins' } */
+    special;
+
+    get type() {
+        let type;
+        type = this.channel ? "channel" : type;
+        type = this.partner ? "partner" : type;
+        type = this.special ? "special" : type;
+        return type;
+    }
```

Comment:
This ternary doesn't feel useful to me, they make the code harder for no reason. I would just use if and early return.
```suggestion
    get type() {
        if (this.channel) {
            return "channel";
        }
        if (this.partner) {
            return "partner";
        }
        if (this.special) {
            return "special";
        }
        throw new Error("Unkown mention type");
    }
```